xbmc/favourites/FavouritesService.cpp

```diff
@@ -237,7 +232,19 @@ bool CFavouritesService::AddOrRemove(const CFileItem& item, int contextWindow)
 bool CFavouritesService::IsFavourited(const CFileItem& item, int contextWindow) const
 {
   std::unique_lock<CCriticalSection> lock(m_criticalSection);
-  return m_favourites.Contains(GetFavouritesUrl(item, contextWindow));
+
+  const CFavouritesURL favURL{item, contextWindow};
+  for (const auto& favItem : m_favourites)
```

Comment:
I know this is nothing you introduced but would be highly preferable that we could simply set a property on the item if it is favourites to avoid the lookup.

Reply:
Hmm, to set the property you would have to look it up here once, so this would be only some caching for the lifetime of the item instance and I'm not sure whether we will have subsequent requests for the favourite check at the same instance at all.

I do agree such a property would be great, but lacking ideas how to implement this kodi-wide, for all kind of items.